arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/test/Laws.kt

```diff
@@ -23,15 +25,14 @@ fun StringSpec.testLaws(vararg laws: List<Law>): Unit = laws
   .distinctBy { law: Law -> law.name }
   .forEach { law: Law ->
     addTest(TestName(null, law.name, false), false, null) {
-      law.test(StringSpecScope(this.coroutineContext, testCase))
+      runTest { law.test(this) }
     }
   }
 
-fun StringSpec.testLaws(prefix: String, vararg laws: List<Law>): Unit = laws
-  .flatMap { list: List<Law> -> list.asIterable() }
-  .distinctBy { law: Law -> law.name }
-  .forEach { law: Law ->
-    addTest(TestName(prefix, law.name, false), false, null) {
-      law.test(StringSpecScope(this.coroutineContext, testCase))
-    }
-  }
+fun testLawsCommon(lawSet: LawSet) = testLawsCommon(lawSet.laws)
+fun testLawsCommon(vararg laws: List<Law>) = runTest {
+  laws
+    .flatMap(List<Law>::asIterable)
+    .distinctBy(Law::name)
+    .forEach { law: Law -> law.test(this) }
```

Comment:
```suggestion
    .forEach { law: Law -> law.test() }
```

Reply:
I'm only wandering here if we should enhance the assertion message now, since it's all running as a single test now it will be impossible to figure out which failed assertions came from which `Law`.

```kotlin
runCatching { law.test() }
  .onFailure { throw AssertionError(law.name, it) }
```

Reply:
Since the associated ticket didn't mention replacing the kotest assertions themselves, I added a `withClue` block which works cross-platform and now reads:

On JVM:

```
Caused by java.lang.AssertionError: In MonoidLaws(name=Option, empty=Option.None, combine=(arrow.core.Option<kotlin.Int>, arrow.core.Option<kotlin.Int>) -> arrow.core.Option<kotlin.Int>, GEN=io.kotest.property.arbitrary.TrampolineArb@2fb1ecf1, eq=(F, F) -> kotlin.Boolean)
Testing Law(name=Semigroup Laws (Option): associativity, test=() -> kotlin.Unit)
Found Option.Some(877965308) but expected: Option.Some(-2125984056) at
```

On JS:
```
Caused by AssertionError: Testing Law(name=Semigroup Laws (Option): associativity, test=function ($completion) {
      return i.invoke_pbb09x_k$($completion);
    })
Found Option.Some(-1) but expected: Option.Some(1)
```

(it seems like I can't run native tests at the moment, the executable complains about missing `libcrypt.so.1`)